### PR TITLE
Registration routine and uppercase Keyboard extension

### PR DESCRIPTION
**Pull request by ewerspej**
In short: Added registration routine with custom HiddenPinEntry class and extended PINInputType to allow all-uppercase keyboard.

**TL;DR**
Registration routines are common practice with MAUI controls. This simplifies the usage and reduces code repetition. Also, in this case, it prevents registering a shared custom handler for all Entry instances. Instead, a custom Entry class called `HiddenPinEntry` was introduced so that the custom handler only applies to instances of that specific type. Registration works simply by adding the following call in *MauiProgram.cs*:

```c#
var builder = MauiApp.CreateBuilder();

builder
    .UseMauiApp<App>()
    .UsePinView(); // this will register the required handler(s)

return builder.Build();
```

The PINInputType enum was extended with an `AlphaNumericUppercase` value so that an all-uppercase keyboard can be used. This is useful when alphanumeric PINs and access codes only allow uppercase characters.

### CODE REVIEW

**Comment by MGohil** [comment]
@ewerspej : Thank you for your PR to add this Uppercase input type and few other code improvements. I appreciate your efforts in this.

**Comment by ewerspej** [comment]
@MGohil You're more than welcome 🤝
